src/explore-education-statistics-common/src/services/util/combineMeasuresWithDuplicateLocationCodes.ts

```diff
@@ -0,0 +1,245 @@
+import {
+  LocationOption,
+  TableDataResult,
+} from '@common/services/tableBuilderService';
+import {
+  groupBy,
+  isEqual,
+  mapValues,
+  partition,
+  sum,
+  uniq,
+  uniqWith,
+} from 'lodash';
+import produce from 'immer';
+
+type LocationGroupingKey = {
+  level: string;
+  code: string;
+};
+
+type MeasurementsMergeStrategy = (
+  measurementValues: (string | undefined)[],
+) => string;
+
+/**
+ * Combine rows of {@param results} that reference Locations that use duplicate Codes but have unique Names.
+ *
+ * This is possible when, say, a Provider changes its name during the course of the period of time that a Data Table
+ * covers, thus leading to 2 distinct names for the same Provider (under the same Code).
+ *
+ * The API already handles this scenario by returning a single combined Location option which groups these 2 or more
+ * duplicate Locations into a single selectable option in the Table Tool (with the label of the form
+ * "Location A / Location B").
+ *
+ * {@param deduplicatedLocations} is an array of Locations available for the given Subject being queried, and from this
+ * list, the single combined Location label can be determined for use as the Location label for the row header that
+ * these combined result rows will fall under.
+ *
+ * ${@param measurementsMergeStrategy} is the strategy whereby multiple values from the various duplicate Locations
+ * are merged together so that the final value will reside within a single table cell.  By default this is by summing
+ * any numerical values together or, if no numeric values exist, to display the first non-numeric string.
+ */
+export default function combineMeasuresWithDuplicateLocationCodes(
+  results: TableDataResult[],
+  deduplicatedLocations: LocationOption[],
+  measurementsMergeStrategy: MeasurementsMergeStrategy = sumNumericValuesMergeStrategy,
```

Comment:
Is it necessary to have this as combineMeasuresWithDuplicateLocationCodes is only called in one place and uses the default strategy?

Reply:
See other comment about trying to make it more obvious in the tests what was going on :)